Update gemspec dependencies for ruby 3.3

kbrock opened the pull request:
ruby 3.3 deprecation warnings

jrafanie commented on CHANGELOG.md:
```diff
@@ -6,6 +6,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+
+- Explicitly added ostruct dependency since it is leaving the standard library.
+- Explicitly added csv dependency. It is required by prawn-table.
+- Explicitly added matrix. It is required by prawn. (We're not ready to upgrade to 2.5 yet)
```
@kbrock for csv and matrix, why do we need to add indirect dependencies as direct dependencies?  Is there a deprecation warning this addresses?

I would have thought prawn-table and prawn should state their dependencies.  What am I missing?

kbrock replied:
@jrafanie prawn-table hasn't been updated in years, and I am pessimistic that we'll get this change in there.

Prawn has been updated. The best approach is to upgrade the dependency but it was failing the build so I took the easy way out.

https://github.com/prawnpdf/prawn/commit/3658d5125c3b20eb11484c3b039ca6b89dc7d1b7

I created a PR for prawn-table

 https://github.com/prawnpdf/prawn-table/pull/161

jrafanie replied:
Can you fixup the commit message with that info?  Note the typo too:
`Ostrinct dependency now explicit`.  I'm not sure what `declaring` means for the other commit.

jrafanie replied:
I'm ok with adding indirect dependencies to correct a problem in a direct dependency but we should make this very obvious so we can undo this when/if this is ever fixed.  Please comment on the PRs so we can follow up sometime later to see if they have fixed and released it.

jrafanie replied:
Should we bump the VERSION file to match this entry?  Does it make sense to bump minor version due to the dependency changes? It always feels odd when dependencies change in a patch release.

jrafanie replied:
@kbrock Can you bump the VERSION file to 1.8.1?

kbrock committed:
declaring indirect dependencies

Not optimal, but prawn-table has not been updated in a while.
prawn core has been updated for this change, but we are not ready to upgrade at this time.

kbrock commented:
@jrafanie thanks - future me thanks you a lot.

update:
- updated commit message to be much more clear
- updated gemspec to properly state dependencies and when we can remove this indirect dependency.

kbrock commented:
update:
- fixed the ostruct typo

kbrock committed:
Make Ostruct dependency explicit

This change removes a deprecation warning in ruby 3.2
This is required by ruby 3.4 since it will no longer be in standard library

kbrock commented:
update:
- update version.rb
- changed date

jrafanie reviewed (approved):
LGTM
